tests/v2/validation/provisioning/k3s/custom_cluster_test.go

```diff
@@ -221,13 +221,24 @@ func (c *CustomClusterProvisioningTestSuite) testProvisioningK3SCustomCluster(cl
 	err = wait.WatchWait(result, checkFunc)
 	assert.NoError(c.T(), err)
 	assert.Equal(c.T(), clusterName, clusterResp.ObjectMeta.Name)
+	assert.Equal(c.T(), kubeVersion, cluster.Spec.KubernetesVersion)
+
+	clusterIDName, err := clusters.GetClusterIDByName(c.client, clusterName)
+	assert.NoError(c.T(), err)
+
+	err = nodestat.IsNodeReady(client, clusterIDName)
+	require.NoError(c.T(), err)
 
 	clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
 	require.NoError(c.T(), err)
 	assert.NotEmpty(c.T(), clusterToken)
 
-	if harden.Hardened && kubeVersion < "v1.25.0" {
-		err = hardening.HardeningNodes(client, harden.Hardened, nodes, nodesAndRoles)
+	podResults, podErrors := pods.StatusPods(client, clusterIDName)
+	assert.NotEmpty(c.T(), podResults)
+	assert.Empty(c.T(), podErrors)
+
+	if c.hardened && kubeVersion < "v1.25.0" {
```

Comment:
Does this actually work? In terms of the `kubeVersion < "v1.25.0" ` golang has a semantic versioning package as well https://pkg.go.dev/golang.org/x/mod/semver. I'm very curious about that.

Reply:
Nevermind, Ugur confirmed that it does work.

Reply:
I spoke too soon, and found a case where this will be false:

`kubeVersion = v1.25.0+k3s1`. This should be equal yet it'll be false according to this statement.

Reply:
Hmm. In my local/automated testing, I tested with the versions found in Rancher and it always seemed good, but Ugur brings up a good point. My question here though is would we need to make that adjustment?

I had just assumed we'll be referencing the supported Rancher versions, which in that case this check shouldn't be an issue. Correct me if I'm wrong in my assumption though, open to any change.

Reply:
So in my opinion, there's one question for semver vs string comparison: Is the hardening part of the test gonna be used within the release cycles, is/can it support future kubernetes version testing beforehand the release cycles? Or is it gonna be a test that's gonna be updated to support the newly added kubernetes versions?

I definitely see your point on the supported version, Markus, I think this then needs to be a string enum in maybe f.e. `provisioning/config.go` so it'd be type-safe and used in a single place. As something like this (naming as a placeholder, can be anything as you see fit both for file -- can be in extensions too -- and variables names):

```
type MyType string
HardenedSupportVersion MyType = mysemver or string version
```

Reply:
I see, I'll make that adjustment!

Reply:
So that test conditional won't change as K8s 1.25+ doesn't use PSPs. So eventually what we'll need to do is update the hardening tests to account for K8s versions 1.25+, but there's not a path carved yet on how to harden K8s 1.25+ clusters yet.

Once that becomes clear, then we can update the hardening tests.

Reply:
Thank you for the information, Markus! In my opinion, the enum string should be sufficient for now until we get a generic solution for it. What do you guys think?

Reply:
I agree with this I think having an enum string is appropriate rather than a hard coded string. Especially since this will be for all cluster types correct?

Reply:
Yeah, this will be for all cluster types. sounds good to me, I'll go ahead and make that change 👍🏿

Reply:
Resolved.